lib/rubocop/cop/mixin/negative_conditional.rb

```diff
@@ -6,15 +6,20 @@ module Cop
     # Some common code shared between FavorUnlessOverNegatedIf and
     # FavorUntilOverNegatedWhile.
     module NegativeConditional
+      def self.included(mod)
+        mod.def_node_matcher :double_negative?, '(send (send _ :!) :!)'
+      end
+
       def check_negative_conditional(node)
         condition, _body, _rest = *node
 
         # Look at last expression of contents if there are parentheses
         # around condition.
         condition = condition.children.last while condition.type == :begin
-        return unless condition.type == :send
+        return unless condition.type == :send && !double_negative?(condition)
 
         _object, method = *condition
+
         return unless method == :! && !(node.loc.respond_to?(:else) &&
```

Comment:
Hmmm I think we still need this check to see if the only operator is `!`. the additions here only rule out a double negation, which wouldn't be enough to check for this rule.